### ISSUE TITLE
`strptime`: `%U/%W` are not used without day of week

### ISSUE THREAD

**Issue by 67dec636-b907-4a4a-aa3a-f2c426fc6d96** [python/cpython#74985]
BPO | [30802](https://bugs.python.org/issue30802)
--- | :---
Nosy | @musically-ut, @matheusportela, @pganssle, @nooperpudd, @tirkarthi, @serge-bon
PRs | <li>python/cpython#11594</li><li>python/cpython#11594</li><li>python/cpython#11594</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-06-29.08:58:08.577>
labels = ['3.7', '3.8', 'type-bug', 'library']
title = "datetime.datetime.strptime('200722', '%Y%U')"
updated_at = <Date 2019-05-22.13:43:18.013>
user = 'https://github.com/nooperpudd'
```

bugs.python.org fields:
```python
activity = <Date 2019-05-22.13:43:18.013>
actor = 'p-ganssle'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2017-06-29.08:58:08.577>
creator = 'winton'
dependencies = []
files = []
hgrepos = []
issue_num = 30802
keywords = ['patch', 'patch', 'patch']
message_count = 11.0
messages = ['297265', '297266', '297274', '297714', '298840', '299114', '299124', '299446', '332399', '332400', '332456']
nosy_count = 7.0
nosy_names = ['mrabarnett', 'musically_ut', 'matheus.v.portela', 'p-ganssle', 'winton', 'xtreak', 'serge-bon']
pr_nums = ['11594', '11594', '11594']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue30802'
versions = ['Python 2.7', 'Python 3.7', 'Python 3.8']
```

</p></details>

**Comment by 67dec636-b907-4a4a-aa3a-f2c426fc6d96** [comment]
datetime stiptime error according to "%U" format.

Python version: 3.4.3, 2.7.13 ,3.6.0, 3.5.2


```py
from datetime import datetime
```


input:
  datetime.strptime('201726', '%Y%U')

output:
  datetime.datetime(2007, 1, 1, 0, 0)

**Comment by 67dec636-b907-4a4a-aa3a-f2c426fc6d96** [comment]
the output:
   datetime.datetime(2017, 1, 1, 0, 0)

I question about the week format if without the week of the weekday, the datetime should be the fist day of week as the default?

**Comment by 39d85a87-36ea-41b2-b2bb-2be43abb500e** [comment]
Expected result is datetime.datetime(2017, 6, 25, 0, 0).

**Comment by a24690b2-760c-4b16-8d68-d9a44e29742c** [comment]
This case is explicitly mentioned in the documentation: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior


> 7. When used with the strptime() method, %U and %W are only used in calculations when the day of the week and the calendar year (%Y) are specified.

The documentation also says that the %u (day of the week) format specifier was added only in Python 3.6.

~
ut

**Comment by ac0fe71c-dd58-48d3-a60d-861ac82a4ad5** [comment]
Therefore, should we implement the behavior of assuming the first day of the week if none is provided or not?

**Comment by a24690b2-760c-4b16-8d68-d9a44e29742c** [comment]
@Matheus: What other languages support this? Is this (i.e. defaulting to Sunday, or Monday?) the default behavior there?

**Comment by 39d85a87-36ea-41b2-b2bb-2be43abb500e** [comment]
I think the relevant standard is ISO 8601:

https://en.wikipedia.org/wiki/ISO_8601

The first day of the week is Monday.

Note particularly the examples it gives:

    Monday 29 December 2008 is written "2009-W01-1"
    Sunday 3 January 2010 is written "2009-W53-7"

So the first few days of January can be in the last week of the previous year!

**Comment by ac0fe71c-dd58-48d3-a60d-861ac82a4ad5** [comment]
Agreed that we should attain to ISO 8601, as it solves two questions that occurred in this thread:

- According to the specification, it is clear that a week must start on Monday and end on Sunday. Hence datetime.strptime('201726', '%Y%U') could be safely assigned to 2017/06/26 without ambiguity

- It also makes clear on what to do when a week overlaps two years.

IMHO there is no reason to not implement this feature and remove the requirement of using %U with the year and day of the week.

**Comment by Nad834** [comment]
"""
 
 "_environ_"
 "_CGI.py_"
 "_Start_Response_"
 "_Response_Header_"
 "_Header_Name_"
 "_Header_Value_"
 "_Header_Name_" == "_ahmedsoltan.abomariam@gmail.com_"
 "_Start_Response_"
 "_CGI-999_“
 "_call_"
 "_call_str_"
 "_call_String_"
 "_Str_byte_"
 "_String_Bytes_"
 "_String_Type_"
 "_os.environ_"
 ”_input.info_”
 “_String_type_" == "_01112888935_"
 "_Strt_Response_"
 "_sys.info_"
 "_exc.info_"
 "_Response_Headers_"
 "_Header_Name_"
 "_Header_Value_"
 "_Type_String_"
 "_write_( )_""
 "_Write_(String-Content)_"
 "_import_snake.yml_"
 "_install_($176jbnpv6c^dj^&dcy#fb0=-
d81gej5w5zum1f20Ce=w+_w5u)_"
"""
"_if_code_change_" == "_excute_"
"_excute_All_Contribution_"
"_Contribute_Header_"
"_getttribute_"'$176jbnpv6c^dj^&dcy#fb0=-
d81gej5w5zum1f20Ce=w+_w5u
"_Attribute_only_ahmedsoltan.abonariam@gmail.com_"
"_var_immediatly_"
"_Trolling_Explicity_"
"_def_(Date)_"
"_(Date)_" == (05/29/2024)_"
"_input.info_"
"_main_( )_.._egg\_"


""""

**Comment by Nad834** [comment]
"""
 
 "_environ_"
 "_CGI.py_"
 "_Start_Response_"
 "_Response_Header_"
 "_Header_Name_"
 "_Header_Value_"
 "_Header_Name_" == "_ahmedsoltan.abomariam@gmail.com_"
 "_Start_Response_"
 "_CGI-999_“
 "_call_"
 "_call_str_"
 "_call_String_"
 "_Str_byte_"
 "_String_Bytes_"
 "_String_Type_"
 "_os.environ_"
 ”_input.info_”
 “_String_type_" == "_01112888935_"
 "_Strt_Response_"
 "_sys.info_"
 "_exc.info_"
 "_Response_Headers_"
 "_Header_Name_"
 "_Header_Value_"
 "_Type_String_"
 "_write_( )_""
 "_Write_(String-Content)_"
 "_import_snake.yml_"
 "_install_($176jbnpv6c^dj^&dcy#fb0=-
d81gej5w5zum1f20Ce=w+_w5u)_"
"""
"_if_code_change_" == "_excute_"
"_excute_All_Contribution_"
"_Contribute_Header_"
"_getttribute_"'$176jbnpv6c^dj^&dcy#fb0=-
d81gej5w5zum1f20Ce=w+_w5u
"_Attribute_only_ahmedsoltan.abonariam@gmail.com_"
"_var_immediatly_"
"_Trolling_Explicity_"
"_def_(Date)_"
"_(Date)_" == (05/29/2024)_"
"_input.info_"
"_main_( )_.._egg\_"


""""

**Comment by StanFromIreland** [comment]
This is a duplicate of https://github.com/python/cpython/issues/88421. (%W without weekdays, but they are considered together here)

This behavior is consistent with the C implementation and is in the docs. I see no reason to change anything as it may break stuff and is unnecessarily inconsistent with the C.

**Comment by StanFromIreland** [comment]
Changing to *type-feature* as current behavior is correct per current doc:

> When used with the [strptime()](https://docs.python.org/3/library/datetime.html#datetime.datetime.strptime) method, %U and %W are only used in calculations when the day of the week and the calendar year (%Y) are specified.
